### ISSUE TITLE
Replace powermock dependency in pass-authz to enable migration to Java 11

### ISSUE THREAD

**Issue by jabrah** [eclipse-pass/main#266]
It looks like [`powermock`](https://github.com/powermock/powermock) in [`pass-authz`](https://github.com/eclipse-pass/pass-authz) requires a command line argument in order to compile with JDK 11. This workaround, however, will no longer work in JDK 17+, as I believe the argument will be removed.

``` xml
<plugin>
    <groupId>org.apache.maven.plugins</groupId>
    <artifactId>maven-surefire-plugin</artifactId>
    <configuration>
        <argLine>
             --illegal-access=permit
        </argLine>
    </configuration>
</plugin>
```

This workaround is proposed in https://github.com/eclipse-pass/pass-authz/pull/173 to enable migrating to JDK 11 immediately, allowing time to update the tests later.

This library is used in one test class, so it shouldn't be too difficult to replace: 
https://github.com/eclipse-pass/pass-authz/blob/c13bc163cf232788f6e4dc783df06989a17c6769/pass-authz-roles/src/test/java/org/dataconservancy/pass/authz/roles/PassRolesFilterTest.java#L101-L102

We may just be able to replace the use of powermock here with mockito v3+, assuming upgrading mockito doesn't break other things

**Comment by markpatton** [comment]
This should be fixed in #334

**Comment by markpatton** [comment]
No longer relevant.
